--- spring-petclinic-visits-service/src/main/java/org/springframework/samples/petclinic/visits/web/VisitResource.java
+++ spring-petclinic-visits-service/src/main/java/org/springframework/samples/petclinic/visits/web/VisitResource.java
@@ -70,7 +70,7 @@
     }
 
     @GetMapping("pets/visits")
-    public Visits read(@RequestParam("petId") List<Integer> petIds) {
+    public Visits read(@RequestParam("petId") List<Integer> petIds, @RequestParam(value="includeDetails", required=true) boolean includeDetails) {
         final List<Visit> byPetIdIn = visitRepository.findByPetIdIn(petIds);
         return new Visits(byPetIdIn);
     }
